Document jslib-aws v0.12.1 (#1585)
grafana/k6-docs: docs/sources/next/javascript-api/jslib/aws/LambdaClient/_index.md

---
title: 'LambdaClient'
head_title: 'LambdaClient'
description: 'LambdaClient allows interacting with the AWS Lambda service'
weight: 00
---

# LambdaClient

{{< docs/shared source="k6" lookup="blocking-aws-blockquote.md" version="<K6_VERSION>" >}}

> `LambdaClient` interacts with the AWS Lambda. With it, you can invoke a lambda function.

Reviewer: ```suggestion
`LambdaClient` interacts with [AWS Lambda](https://aws.amazon.com/lambda/). With it, you can invoke a Lambda function.
```

Both the dedicated `lambda.js` jslib bundle and the all-encompassing `aws.js` bundle include the `LambdaClient`.

### Methods

| Function                                                                                                                  | Description                     |
| :------------------------------------------------------------------------------------------------------------------------ | :------------------------------ |

> | [invoke(name, payload, [options])](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/jslib/aws/lambdaclient/invoke) | Invokes an AWS lambda function. |

Reviewer: ```suggestion
| [invoke(name, payload, [options])](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/jslib/aws/lambdaclient/invoke) | Invokes an AWS Lambda function. |
```

### Throws

`LambdaClient` methods will throw errors in case of failure.

| Error                 | Condition                                                   |
| :-------------------- | :---------------------------------------------------------- |

> | InvalidSignatureError | when invalid credentials were provided.                     |
> | LambdaInvocationError | when AWS replied to the requested invokation with an error. |
> 

Reviewer: ```suggestion
| InvalidSignatureError | When invalid credentials were provided.                     |
| LambdaInvocationError | When AWS replied to the requested invocation with an error. |

```
